crates/nu-protocol/src/shell_error.rs

```diff
@@ -621,32 +602,14 @@ Either make sure {0} is a string, or add a 'to_string' entry for it in ENV_CONVE
     /// This is a generic error type used for different situations.
     #[error("{0}")]
     #[diagnostic()]
-    SpannedLabeledError(String, String, #[label("{1}")] Span),
-
-    /// This is a generic error type used for different situations.
-    #[error("{0}")]
-    #[diagnostic(help("{3}"))]
-    SpannedLabeledErrorHelp(String, String, #[label("{1}")] Span, String),
-
-    /// This is a generic error type used for different situations.
-    #[error("{0}")]
-    #[diagnostic()]
-    SpannedLabeledErrorRelated(
+    GenericError(
```

Comment:
Most of the diffs involve this particular change: I've coalesced all the various generic errors into a single `GenericError` type that you can dynamically pass all the various fields to.

There are currently 309 of these being used throughout the codebase. Usage is intentionally left suboptimal because, tbh, `GenericError` usage should be limited/reduced over time and moved into more specific errors with unique error codes (and docstrings).